editor/src/clj/editor/editor_extensions.clj

```diff
@@ -154,6 +155,56 @@
       (and (not (resource/resource? node-id-or-resource))
            (some? (graph/ext-lua-value-setter node-id-or-resource property rt project evaluation-context))))))
 
+(def ^:private created-resources-coercer
+  (coerce/vector-of
+    (coerce/one-of
+      graph/resource-path-coercer
+      (coerce/hash-map :req {1 graph/resource-path-coercer}
+                       :opt {:content coerce/string}
```

Comment:
Is creating resources from string content really the preferred way to do it? I suppose it's useful, but it's really annoying to create files with resources embedded as strings like this. Maybe we could rename the key to `text`, and reserve `content` for something like a Lua table representation like the Outline attachment syntax we have for adding stuff to an existing resource?

Reply:
We could add `:properties` later if needed, which would be a more descriptive name.

Reply:
Regarding usefulness, it's very convenient, e.g., for creating resources of custom type, e.g. see the trenchfold script:
https://github.com/astrochili/defold-trenchfold/blob/master/trenchfold/trenchfold.lua

I initially wanted to support property tables, but then I realized it's not only too hard to make it work with the editor sync model, but even the way to do it can already be replicated if we don't support setting properties in this function.

Creating files requires resource sync: it's not undoable. So, an API that accepts properties would have to create files, then make a transaction, and then do a save. This could be done anyway, since the script can transact and save.